src/TreeView/utils.ts

```diff
@@ -0,0 +1,291 @@
+      ) {
+        break;
+      }
+      const enabledChildren = data[parent].children.filter(
+        (x) => !disabledIds.has(x)
+      );
+      if (enabledChildren.length === 0) break;
+      const some = enabledChildren.some((x) => selectedIds.has(x));
+      if (!some) {
+        changes.none.add(parent);
+      } else {
+        if (enabledChildren.every((x) => selectedIds.has(x))) {
+          changes.every.add(parent);
+        } else {
+          changes.some.add(parent);
+        }
+      }
+      currentId = parent;
+    }
+  }
+  return changes;
+};
+
+export const getAccessibleRange = ({
+  data,
+  expandedIds,
+  from,
+  to,
+}: {
+  data: INode[];
+  expandedIds: Set<number>;
+  from: number;
+  to: number;
+}) => {
+  const range: number[] = [];
+  const max_loop = Object.keys(data).length;
+  let count = 0;
+  let currentId: number | null = from;
+  range.push(from);
+  if (from < to) {
+    while (count < max_loop) {
+      currentId = getNextAccessible(data, currentId, expandedIds);
+      currentId != null && range.push(currentId);
+      if (currentId == null || currentId === to) break;
+      count += 1;
+    }
+  } else if (from > to) {
+    while (count < max_loop) {
+      currentId = getPreviousAccessible(data, currentId, expandedIds);
+      currentId != null && range.push(currentId);
+      if (currentId == null || currentId === to) break;
+      count += 1;
+    }
+  }
+
+  return range;
+};
+
+interface ITreeNode {
+  name: string;
+  children?: ITreeNode[];
+}
+
+export const flattenTree = function(tree: ITreeNode): INode[] {
+  let count = 0;
+  const flattenedTree: INode[] = [];
+
+  const flattenTreeHelper = function(tree: ITreeNode, parent: number | null) {
+    const node: INode = {
+      id: count,
+      name: tree.name,
+      children: [],
+      parent,
+    };
+    flattenedTree[count] = node;
+    count += 1;
+    if (tree.children == null || tree.children.length === 0) return;
+    for (const child of tree.children) {
+      flattenTreeHelper(child, node.id);
+    }
+    node.children = flattenedTree
+      .filter((x) => x.parent === node.id)
+      .map((x: INode) => x.id);
+  };
+
+  flattenTreeHelper(tree, null);
+  return flattenedTree;
+};
+
+export const getAriaSelected = ({
+  isSelected,
+  isDisabled,
+  multiSelect,
+}: {
+  isSelected: boolean;
+  isDisabled: boolean;
+  multiSelect: boolean;
+}) => {
+  if (isDisabled) return undefined;
+  if (multiSelect) return isSelected;
+  return isSelected ? true : undefined;
+};
+
+export const propagatedIds = (
+  data: INode[],
+  ids: number[],
+  disabledIds: Set<number>
+) =>
+  ids.concat(
+    ...ids
+      .filter((id) => isBranchNode(data, id))
+      .map((id) => getDescendants(data, id, disabledIds))
+  );
+
+const isIE = () => window.navigator.userAgent.match(/Trident/);
+
+export const onComponentBlur = (
+  event: React.FocusEvent,
+  treeNode: HTMLUListElement | null,
+  callback: () => void
+) => {
+  if (treeNode == null) {
```

Comment:
@lissitz It's possible to have a null ref passed to this function for the `treeNode` prop. I'm logging a warning to the console unless there is something more preferable.

CC: @dgreene1